### PR TITLE
fix: Load multi-trial learning curve with only training metrics [WEB-1026]

### PR DESCRIPTION
**Pull request by mapmeld**
## Description

LearningCurve typically is based on its searcher metric from the validation metrics. This causes a delay if validation hasn't happened and filled metric names yet.

Solution: similar to our single-trial charts, if our preferred metric is not available yet, show the 0th metric.

## Test Plan

- Visit a previous multi-trial experiment and it should show validation metric.
- Create a test experiment with no validation metrics, or in `src/hooks/useMetricNames.ts`, simulate by commenting out

```typescript
        (event.validationMetrics || [])
          .filter((metric) => !xAxisMetrics.includes(metric))
          .forEach((metric) => (validationMetricsMap[metric] = true));
```

- With validation metrics removed, a multi-trial experiment will show one of its training metrics.
- Fork a multi-trial experiment and it will start out blank. When metric names updates (can take 15-30 seconds after the first recorded metrics) then the chart will appear with the training metric. 
- Fork a multi-trial experiment and once it is active/running, refresh repeatedly; as soon as metrics are recorded it will show the training metric.
- Learning Curve metric filter is not persisted (when you return after validation metrics are recorded or re-enabled, it will prefer the searcher metric over a previous selection)

## Checklist

- [x] Changes have been manually QA'd
- [x] User-facing API changes need the "User-facing API Change" label.
- [x] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [x] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

**Review by garrett361 — COMMENTED** [review]
I am seeing `No learning curve data to show.` when submitting a sample multi-Trial Experiment which only reports training metrics.  Testing in `devcluster`.

Using the `random.yaml` config [in this link](https://github.com/garrett361/determined/tree/scratchwork/examples/scratchwork/visualize_late_or_no_val_metrics_test/no_val_metrics) to test.

**Comment by garrett361** [comment]
Cool, after the last push I am seeing multi-Trial Experiments with no validation metrics visualized in `devcluster`. Very nice!

File: master/internal/db/postgres_experiments.go [hunk 1]
```diff
@@ -355,10 +355,10 @@ SELECT t.id FROM (
   SELECT t.id,
     %s((v.metrics->'validation_metrics'->>$1)::float8) as best_metric
   FROM trials t
-  JOIN validations v ON t.id = v.trial_id
+  LEFT JOIN validations v ON t.id = v.trial_id
   WHERE t.experiment_id=$2
   GROUP BY t.id
-  ORDER BY best_metric %s
+  ORDER BY best_metric %s NULLS LAST
```
Comment (trentwatt) [inline master/internal/db/postgres_experiments.go:361]: probably not worth doing training length as a fallback as a part of this PR, but does a secondary sort make sense here? 

maybe `end_time DESC NULLS FIRST` to see the active ones?

**Review by trentwatt — APPROVED** [review]
LGTM, good to have this fixed.

I noticed though if I comment out the `setBatches` in `ExperimentVisualization` the metrics select drop-down doesn't show up (and the page is pretty sparse).

IMO batches shouldn't need to come back with anything for the metric select to show up, but I don't know that this would actually present itself in practice.

**Pull request merged**